addons/crm/crm_lead.py

```diff
@@ -127,214 +100,227 @@ def _read_group_stage_ids(self, cr, uid, ids, domain, read_group_order=None, acc
+            search_domain += ['|', ('id', 'in', self.ids), ('case_default', '=', True)]
         # retrieve type from the context (if set: choose 'type' or 'both')
-        type = self._resolve_type_from_context(cr, uid, context=context)
+        type = self._resolve_type_from_context()
         if type:
             search_domain += ['|', ('type', '=', type), ('type', '=', 'both')]
         # perform search
-        stage_ids = stage_obj._search(cr, uid, search_domain, order=order, access_rights_uid=access_rights_uid, context=context)
-        result = stage_obj.name_get(cr, access_rights_uid, stage_ids, context=context)
+        stage_ids = CrmStage._search(search_domain, order=order, access_rights_uid=access_rights_uid)
+        stage_rec = CrmStage.browse(stage_ids)
+        result = stage_rec.name_get()
         # restore order of the search
         result.sort(lambda x, y: cmp(stage_ids.index(x[0]), stage_ids.index(y[0])))
-
         fold = {}
-        for stage in stage_obj.browse(cr, access_rights_uid, stage_ids, context=context):
+        for stage in stage_rec:
             fold[stage.id] = stage.fold or False
         return result, fold
 
-    def fields_view_get(self, cr, user, view_id=None, view_type='form', context=None, toolbar=False, submenu=False):
-        if context and context.get('opportunity_id'):
-            action = self._get_formview_action(cr, user, context['opportunity_id'], context=context)
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        if self.env.context.get('opportunity_id'):
+            action = self._get_formview_action(self.env.context['opportunity_id'])
             if action.get('views') and any(view_id for view_id in action['views'] if view_id[1] == view_type):
                 view_id = next(view_id[0] for view_id in action['views'] if view_id[1] == view_type)
-        res = super(crm_lead, self).fields_view_get(cr, user, view_id, view_type, context, toolbar=toolbar, submenu=submenu)
+        res = super(CrmLead, self).fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
         if view_type == 'form':
-            res['arch'] = self.fields_view_get_address(cr, user, res['arch'], context=context)
+            res['arch'] = self.fields_view_get_address(res['arch'])
         return res
 
     _group_by_full = {
         'stage_id': _read_group_stage_ids
     }
 
-    def _compute_day(self, cr, uid, ids, fields, args, context=None):
+    @api.multi
+    @api.depends('create_date', 'date_open', 'date_closed')
+    def _compute_day(self):
         """
         :return dict: difference between current date and log date
         """
-        res = {}
-        for lead in self.browse(cr, uid, ids, context=context):
-            for field in fields:
-                res[lead.id] = {}
+        for lead in self:
+            for field in self._fields:
```

Comment:
self._fields ? this one is not same as fields. for self._fields it wil iterate for all fields.. need to calculate for function field only.